homeassistant/components/surepetcare/lock.py

```diff
@@ -0,0 +1,99 @@
+"""Support for Sure PetCare Flaps locks."""
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from surepy.entities import SurepyEntity
+from surepy.enums import EntityType, LockState
+
+from homeassistant.components.lock import STATE_LOCKED, STATE_UNLOCKED, LockEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SurePetcareDataCoordinator
+from .const import DOMAIN
+from .entity import SurePetcareEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Sure PetCare locks on a config entry."""
+
+    entities: list[SurePetcareLock] = []
+
+    coordinator: SurePetcareDataCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    for surepy_entity in coordinator.data.values():
+        if surepy_entity.type in [
+            EntityType.CAT_FLAP,
+            EntityType.PET_FLAP,
+        ]:
+
+            for lock_state in (
+                LockState.LOCKED_IN,
+                LockState.LOCKED_OUT,
+                LockState.LOCKED_ALL,
+            ):
+                entities.append(
+                    SurePetcareLock(surepy_entity.id, coordinator, lock_state)
+                )
+
+    async_add_entities(entities)
+
+
+class SurePetcareLock(SurePetcareEntity, LockEntity):
+    """A lock implementation for Sure Petcare Entities."""
+
```

Comment:
```suggestion
    coordinator: SurePetcareDataCoordinator
```

You might be able to avoid the `cast` with something like this

Reply:
Ahh, nice